lib/jekyll/filters/url_filters.rb

```diff
@@ -76,13 +76,16 @@ def compute_relative_url(input)
 
         parts = [sanitized_baseurl, input]
         Addressable::URI.parse(
-          parts.compact.map { |part| ensure_leading_slash(part.to_s) }.join
+          parts.map! { |part| ensure_leading_slash(part.to_s) }.join
         ).normalize.to_s
       end
 
       def sanitized_baseurl
         site = @context.registers[:site]
-        site.config["baseurl"].to_s.chomp("/")
+        baseurl = site.config["baseurl"]
+        return "" if baseurl.nil?
+
+        baseurl.to_s.chomp("/")
```

Comment:
site.config["baseurl"] isn't going to change so we could memoize this whole method, right?

Reply:
Yes, we could memoize this. Great idea! :+1: